object_model/literal.py

```diff
@@ -17,6 +17,15 @@ def __init__(self, v, name, xp_state):
         self.i = 0
         self.n = 1
 
+        # If no default passed in, set default to the first item in the literal.
+        if default is None:
```

Comment:
I don't think we can have this check here. You always call `forEach` AFTER the constructor, so __oneByOne__ check will always evaluate to false. Need to correct this logic so that the default is correctly set.

Reply:
Hmm, forEach() is only called when the user calls literalForEach(). It would work to set the default of a "foreach" type literal in the forEach method, but what about if the user just creates a normal literal that is not "foreach"? In this case, I think we would have to change the default in the constructor (since foreach isn't called for this type of literal) and we do not want the default to remain None.

Reply:
This is why I decided to always change the default in the constructor - lets talk about this in the meeting!